lib/sda/overlay.go

```diff
@@ -283,7 +280,7 @@ func (o *Overlay) Close() {
 	o.nodeLock.RLock()
 	defer o.nodeLock.RUnlock()
 	for _, n := range o.nodes {
-		if err := n.ProtocolInstance().Shutdown(); err != nil {
+		if err := n.Close(); err != nil {
```

Comment:
Well now `node` has a goroutine, so we need to shut it down correctly, where before we could just tear it down. So instead of `overlay` calling `Shutdown` on `ProtocolInstance`, we have `overlay` calling `Close` on `node`, which will call `Shutdown` on `ProtocolInstance`.

Sheesh, making a long channel seems easier ;)